### ISSUE TITLE
PIE804 raised when providing keywords as kwargs

### ISSUE THREAD

**Issue by 153957** [charliermarsh/ruff#2623]
The following code raises PIE804:
```
def abc(**kwargs):
    xyz(kwargs.get('for', 1))

abc(**{'for': 3})
```

But since `for` is a Python keyword it is not allowed to provide it directly as a keyword:
```
abs(for=3)
```

There are some 'clear' exceptions for PIE804, should these be handled by ruff or should I manually ignore these cases?

`ruff 0.0.243`

**Comment by charliermarsh** [comment]
Yeah I think we should verify that all keys are valid identifiers.
